toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -272,69 +270,60 @@ private void installModule(Module module) {
         return unNamedAndNamedProviders.unNamedProvider;
       }
 
-      Map<String, ScopedProviderImpl<? extends T>> mapNameToProvider = unNamedAndNamedProviders.getMapNameToProvider();
-      if (mapNameToProvider == null) {
+      if (unNamedAndNamedProviders.mapNameToProvider == null) {
         return null;
       }
-      return mapNameToProvider.get(bindingName);
+      return unNamedAndNamedProviders.mapNameToProvider.get(bindingName);
```

Comment:
but all those changes will make more conflicts with the lock free branch. If Unnamed provider could remain mostly unchanged, it would be easier.